### PR TITLE
layer: Use Vulkan-Utility-Libraries for dispatch tables

### PR DESCRIPTION
**Pull request by charles-lunarg**
The Vulkan-Utility-Libraries now provides the necessary dispatch tables for the profiles layer. This removes the dependency on ValidationLayers. To accomplish the switch, the StringAPIVersion function was added locally, since it used to come from the ValidationLayers but is not currently available in the Utility-Libraries.

### CODE REVIEW

File: layer/vk_layer_table.cpp [hunk 1]
```diff
@@ -19,22 +19,26 @@
  * Author: Tobin Ehlis <tobin@lunarg.com>
  */
 #include <assert.h>
+#include <sstream>
+#include <iomanip>
 #include <unordered_map>
-#include "generated/vk_dispatch_table_helper.h"
+#include "vulkan/utility/vul_dispatch_table.h"
```
Comment (christophe-lunarg) [inline layer/vk_layer_table.cpp:25]: Shouldnt it be <vulkan/utility/vul_dispatch_table.h>?

File: layer/vk_layer_table.cpp [hunk 2]
```diff
@@ -120,28 +124,39 @@ VkLayerInstanceDispatchTable *initInstanceTable(VkInstance instance, const PFN_v
...
     return pTable;
 }
 
-VkLayerDispatchTable *initDeviceTable(VkDevice device, const PFN_vkGetDeviceProcAddr gpa) {
+VulDeviceDispatchTable *initDeviceTable(VkDevice device, const PFN_vkGetDeviceProcAddr gpa) {
     return initDeviceTable(device, gpa, tableMap);
 }
+
+// Convert integer API version to a string
+std::string StringAPIVersion(uint32_t version) {
```
Comment (christophe-lunarg) [inline layer/vk_layer_table.cpp:154]: This isn't used anywhere.

**Review by christophe-lunarg — APPROVED** [review]

**Comment by juan-lunarg** [comment]
Looks like the Android build was relying on an implicit include from vvl.

```
/home/runner/work/Vulkan-Profiles/Vulkan-Profiles/layer/profiles_settings.cpp:72:33: error: use of undeclared identifier 'ANDROID_LOG_INFO'
            __android_log_print(ANDROID_LOG_INFO, "Profiles", "%s", message.c_str());
```

The following should fix it.

```
#if defined __ANDROID__
#include <android/log.h>
#endif
```

**Comment by charles-lunarg** [comment]
> The following should fix it.
> 
> ```
> #if defined __ANDROID__
> #include <android/log.h>
> #endif
> ```

Already figured that out as you commented - let me know if where I put the include makes sense.

**Review by juan-lunarg — APPROVED** [review]
💯

**Review by juan-lunarg — COMMENTED** [review]
The BUILD.md still references the validation layers as a dependency.

File: scripts/CMakeLists.txt [hunk 3]
```diff
@@ -118,13 +118,6 @@ if (VULKAN_UTILITY_LIBRARIES_INSTALL_DIR)
     list(APPEND CMAKE_PREFIX_PATH ${VULKAN_UTILITY_LIBRARIES_INSTALL_DIR})
 endif()
 
-# TODO: https://github.com/KhronosGroup/Vulkan-Utility-Libraries/issues/13
-# In the future profiles will NOT depend on the validation layers.
-if (VULKAN_VALIDATIONLAYERS_INSTALL_DIR)
-    list(APPEND CMAKE_INCLUDE_PATH ${VULKAN_VALIDATIONLAYERS_INSTALL_DIR}/include/vulkan)
-    set(CMAKE_INCLUDE_PATH ${CMAKE_INCLUDE_PATH} PARENT_SCOPE)
-endif()
-
```
Comment (juan-lunarg) [inline scripts/CMakeLists.txt:127]: So beautiful to see this gone T_T

**Comment by juan-lunarg** [comment]
> The BUILD.md still references the validation layers as a dependency.

The BUILD.md needs cleanup anyway. That can be done in another PR.

**Pull request merged**